src/lib/sha256.c

```diff
@@ -411,16 +404,21 @@ void ecdsa_sha256_update(ecdsa_sha256_context_t *sc, const void *data, size_t le
 
 		if (sc->bufferLength == 64L) {
 			SHA256Guts(sc, sc->buffer.words);
-			needBurn = 1;
 			sc->bufferLength = 0L;
 		}
 	}
 
 	while (len > 63L) {
+		uint32_t words[64 / sizeof(uint32_t)];
+
 		sc->totalLength += 512L;
 
-		SHA256Guts(sc, data);
-		needBurn = 1;
+		if (IS_ALIGNED(data, sizeof(uint32_t))) {
+			SHA256Guts(sc, data);
+		} else {
+			memcpy(words, data, sizeof(words));
+			SHA256Guts(sc, words);
+		}
```

Comment:
How about copying to the buffer regardless of alignment?

Reply:
My idea was that while no alignment is guaranteed, quite often the passed buffer will be aligned after all - in particular when large blocks of a file are hashed at a time. In these cases avoiding the memcpy seems worthwhile.